### PR TITLE
Fix_issue_41

### PR DESCRIPTION
**Pull request by TiKeil**
Fix for #41 . Test are updated and passing on arch.

### CODE REVIEW

File: dune/xt/la/test/eigensolver_for_real_matrix_with_real_evs_from_3d_pointsource.tpl [hunk 1]
```diff
@@ -195,22 +159,12 @@ TEST_F(EigenSolverForMatrixFrom3dPointsource_{{T_NAME}}, is_constructible)
 
 TEST_F(EigenSolverForMatrixFrom3dPointsource_{{T_NAME}}, gives_correct_eigenvalues)
 {
-  gives_correct_eigenvalues({ {"eigen", "1e-13"}, {"shifted_qr", "1e-14"} });
-}
-
-TEST_F(EigenSolverForMatrixFrom3dPointsource_{{T_NAME}}, gives_correct_eigenvalues_in_correct_order)
-{
-  gives_correct_eigenvalues_in_correct_order({ {"eigen", /*we_expect_a_failure: */ "-1"}, {"shifted_qr", /*we_expect_a_failure: */ "-1"} });
+  gives_correct_eigenvalues({ {"lapack", "1e-14"}, {"eigen", "1e-13"}, {"shifted_qr", "1e-14"}});
```
Comment (tobiasleibner) [inline dune/xt/la/test/eigensolver_for_real_matrix_with_real_evs_from_3d_pointsource.tpl]: You probably also need to add the same tolerances for the numpy solver, as it uses LAPACK internally.

File: dune/xt/la/test/eigensolver_for_real_matrix_with_real_evs_from_3d_pointsource.tpl [hunk 2]
```diff
@@ -195,22 +159,12 @@ TEST_F(EigenSolverForMatrixFrom3dPointsource_{{T_NAME}}, is_constructible)
...
 
 TEST_F(EigenSolverForMatrixFrom3dPointsource_{{T_NAME}}, gives_correct_real_eigenvalues)
 {
-  gives_correct_real_eigenvalues({ {"eigen", "1e-13"}, {"shifted_qr", "1e-14"} });
-}
-
-TEST_F(EigenSolverForMatrixFrom3dPointsource_{{T_NAME}}, gives_correct_real_eigenvalues_in_correct_order)
-{
-  gives_correct_real_eigenvalues_in_correct_order({ {"eigen", /*we_expect_a_failure: */ "-1"}, {"shifted_qr", /*we_expect_a_failure: */ "-1"} });
+  gives_correct_real_eigenvalues({ {"lapack", "1e-14"}, {"eigen", "1e-13"}, {"shifted_qr", "1e-14"} });
```
Comment (tobiasleibner) [inline dune/xt/la/test/eigensolver_for_real_matrix_with_real_evs_from_3d_pointsource.tpl]: See above (add numpy)

**Comment by tobiasleibner** [comment]
Looks good to me, except that you probably need to adjust the tolerances for the numpy solver, too. But you can just wait for travis and see if there are any failing tests.

This pull request includes #36 and #42, so we can close these once this pull request is merged.

**Comment by ftalbrecht** [comment]
As a side note for us in the future: numpy uses lapack internally. So whenever we have to adjust one of the two tolerances, it is likely that the other one has to be adjusted accordingly.

**Comment by tobiasleibner** [comment]
Though travis is green now, please do not merge before @renemilk had a look at the test setup, apparently a lot of the tests that should should be run are not run on travis anymore.

**Comment by renemilk** [comment]
At least the bin count is correct now, yes. https://github.com/dune-community/dune-xt-la/pull/44/files#diff-e8e7f4050a5eea33f8b6708737dd31afR1
If you've added/removed tests you can check yourself if they appear/got removed there too.

**Comment by tobiasleibner** [comment]
The test look fine, so this can finally be merged.

**Comment by ftalbrecht** [comment]
Thx for the effort!

**Pull request merged**